### DISCUSSION TITLE
Close Modal when Clicking Outside the Modal

### DISCUSSION THREAD

**Discussion by alibugrat** [saadeghi/daisyui#646] [Q&A] [answered]
Hi Guys,

Is there any option to close modal when clicking outside the modal?

Thanks.

**Comment by saadeghi** [comment, marked as answer]
There's an example here: https://daisyui.com/components/modal/#modal-that-closes-when-clicked-outside

**Reply by alibugrat** [reply]
I missed. Thanks.

**Comment by iamgenes** [comment]
what if the modal opens by url?

**Comment by nkomp18** [comment]
I don't understand why daisy ui relies on "label" to control the functionality of a modal. It is illegal to use "div" or "form" elements inside a "label" so it's very limiting for the content of the modal. 
Is there a way to do it without the use of "label"?

**Reply by saadeghi** [reply]
Because daisyUI is CSS only. Without JS we can't detect a button click. The only way is using a checkbox + label (because a label can check/uncheck a checkbox) or using anchor links.  

You can also use JS if you want, then add/remove `modal-open` class using JS.  

And... It's not illegal (?) You can use div or form element inside label and it's valid HTML.

**Reply by nkomp18** [reply]
You are right about needing JS to make it work and thanks for clarifying but I will insist that the label element is limiting because it can only contain "Phrasing content" according to the spec: https://html.spec.whatwg.org/multipage/forms.html#the-label-element
Phrasing content is the text of the document, as well as elements that mark up that text at the intra-paragraph level.

If you add a "div" or other types of elements under a label, it may work in some browsers but you get a warning. 
Since it does work for "label", I wonder whether there's a way to also make it work for div without the need a js.

**Reply by saadeghi** [reply]
As far as I know, another way to do this without JS would be using a separate element for backdrop (a `<label>` or `<a>` anchor link) and visually put it under the modal. This way, it can be closed by clicking outside and we won't need to put the whole modal content inside the `<label>`  
The above idea would be a breaking change for existing daisyUI users of course.

Also, another option which I'm working on for daisyUI `3.0` is using the [almost] new `<dialog>` native HTML element instead of the current checkbox method. The challenge is, `<dialog>` needs JS to open/close so I'm trying to find a way to make it work without JS (at least visually)

**Reply by alanhamlett** [reply]
Looks like you got `<dialog>` to work with Daisy's Modal? I'm using `<dialog>` but can't find docs on closing it on outside click.

**Reply by saadeghi** [reply]
@alanhamlett https://daisyui.com/components/modal/#dialog-modal-closes-when-clicked-outside
